src/simple_rules.ml

```diff
@@ -66,11 +66,7 @@ let copy_files sctx ~dir ~scope ~src_dir (def: Copy_files.t) =
     let src_glob = SC.expand_vars_string sctx ~dir def.glob ~scope in
     Path.relative src_dir src_glob ~error_loc:loc
   in
-  (* The following condition is required for merlin to work.
-     Additionally, the order in which the rules are evaluated only
-     ensures that [sources_and_targets_known_so_far] returns the
-     right answer for sub-directories only. *)
-  if not (Path.is_descendant glob_in_src ~of_:src_dir) then
+  if def.syntax_version < (1, 3) && not (Path.is_descendant glob_in_src ~of_:src_dir) then
```

Comment:
Just a note that we usually wrap our code to 80 chars. I'm not sure if it's over the limit here, but something to keep in mind.